parquet-hadoop/src/test/java/org/apache/parquet/encodings/TestTypeEncodings.java

```diff
@@ -0,0 +1,417 @@
+        { PrimitiveTypeName.DOUBLE },
+        { PrimitiveTypeName.BINARY },
+        { PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY }
+    });
+  }
+
+  public TestTypeEncodings(PrimitiveTypeName typeName) {
+    this.paramTypeName = typeName;
+  }
+
+  @BeforeClass
+  public static void initialize() throws IOException {
+    Random random = new Random(RANDOM_SEED);
+    intGenerator = new RandomValues.IntGenerator(random.nextLong());
+    longGenerator = new RandomValues.LongGenerator(random.nextLong());
+    int96Generator = new RandomValues.Int96Generator(random.nextLong());
+    floatGenerator = new RandomValues.FloatGenerator(random.nextLong());
+    doubleGenerator = new RandomValues.DoubleGenerator(random.nextLong());
+    binaryGenerator = new RandomValues.BinaryGenerator(random.nextLong());
+    fixedBinaryGenerator = new RandomValues.FixedGenerator(random.nextLong(), FIXED_LENGTH);
+  }
+
+  @Test
+  public void testTypeEncodings() throws Exception {
+    int pageSize = 64; // 64B
+    int rowGroupSize;
+
+    // Let's use a smaller row group size for booleans so that it span
+    // many pages.
+    if (this.paramTypeName == PrimitiveTypeName.BOOLEAN) {
+      rowGroupSize = 32 * 1024;  // 32K
+    } else {
+      rowGroupSize = 256 * 1024; // 256K
+    }
+
+    Path parquet1 = createTempFile();
+    Path parquet2 = createTempFile();
+
+    System.out.println(String.format("Testing %s encodings using ROW_GROUP_SIZE=%d PAGE_SIZE=%d",
+        this.paramTypeName.toString(), rowGroupSize, pageSize));
+
+    List<?> randomValues = generateRandomValues(this.paramTypeName, RECORD_COUNT);
+
+    writeValuesToFile(parquet1, this.paramTypeName, randomValues, rowGroupSize, pageSize, PARQUET_1_0);
+    writeValuesToFile(parquet2, this.paramTypeName, randomValues, rowGroupSize, pageSize, PARQUET_2_0);
+
+    PageGroupValidator.validatePages(parquet1, randomValues);
+    PageGroupValidator.validatePages(parquet2, randomValues);
+  }
+
+  private Path createTempFile() throws IOException {
+    TemporaryFolder tempFolder = new TemporaryFolder();
+    File file = tempFolder.newFile();
+    file.delete();
+    file.deleteOnExit();
+    return new Path(file.getAbsolutePath());
+  }
+
+  private void writeValuesToFile(Path file, PrimitiveTypeName type, List<?> values, int rowGroupSize, int pageSize, WriterVersion version) throws IOException {
+    MessageType schema;
+    if (type == PrimitiveTypeName.FIXED_LEN_BYTE_ARRAY) {
+      schema = Types.buildMessage().required(type).length(FIXED_LENGTH).named("field").named("test");
+    } else {
+      schema = Types.buildMessage().required(type).named("field").named("test");
+    }
+
+    SimpleGroupFactory message = new SimpleGroupFactory(schema);
+    GroupWriteSupport.setSchema(schema, configuration);
+
+    ParquetWriter<Group> writer = new ParquetWriter<Group>(file, new GroupWriteSupport(),
+        CompressionCodecName.UNCOMPRESSED, rowGroupSize, pageSize, 0, false, false, version, configuration);
```

Comment:
Could you add a comment that calls out the fact that we are avoiding dictionary encoding?